man/qbooblyrics.1
.\" -*- coding: utf-8 -*-
.\" This file was generated automatically by tools/make_man.sh.
.TH QBOOBLYRICS 1 "27 February 2019" "qbooblyrics 1\&.5"
.SH NAME
qbooblyrics \- search lyrics
.SH SYNOPSIS
.B qbooblyrics
[\-h] [\-dqv] [\-b \fIbackends\fR] ...
.br
.B qbooblyrics
[\-\-help] [\-\-version]

.SH DESCRIPTION
.LP

Qt application allowing to search song lyrics.

.SS Supported websites:
* lyricsdotcom (Lyrics.com lyrics website)
.br
* lyricsmode (Lyricsmode.com lyrics website)
.br
* lyricsplanet (Lyricsplanet.com song lyrics website)
.br
* paroles2chansons (Paroles2chansons.com song lyrics website)
.br
* parolesmania (Paroles Mania lyrics website)
.br
* parolesmusique (paroles\-musique lyrics website)
.br
* parolesnet (paroles.net lyrics website)
.SH OPTIONS
.TP
\fB\-\-version\fR
show program's version number and exit
.TP
\fB\-h\fR, \fB\-\-help\fR
show this help message and exit
.TP
\fB\-b BACKENDS\fR, \fB\-\-backends=BACKENDS\fR
what backend(s) to enable (comma separated)
.TP
\fB\-e EXCLUDE_BACKENDS\fR, \fB\-\-exclude\-backends=EXCLUDE_BACKENDS\fR
what backend(s) to exclude (comma separated)
.TP
\fB\-I\fR, \fB\-\-insecure\fR
do not validate SSL
.TP
\fB\-\-nss\fR
Use NSS instead of OpenSSL

.SH LOGGING OPTIONS
.TP
\fB\-d\fR, \fB\-\-debug\fR
display debug messages. Set up it twice to more verbosity
.TP
\fB\-q\fR, \fB\-\-quiet\fR
display only error messages
.TP
\fB\-v\fR, \fB\-\-verbose\fR
display info messages
.TP
\fB\-\-logging\-file=LOGGING_FILE\fR
file to save logs
.TP
\fB\-a\fR, \fB\-\-save\-responses\fR
save every response
.TP
\fB\-\-export\-session\fR
log browser session cookies after login

.SH CONDITION
The \-c and \-\-condition is a flexible way to filter and get only interesting results. It supports conditions on numerical values, dates, durations and strings. Dates are given in YYYY\-MM\-DD or YYYY\-MM\-DD HH:MM format. Durations look like XhYmZs where X, Y and Z are integers. Any of them may be omitted. For instance, YmZs, XhZs or Ym are accepted.
The syntax of one expression is "\fBfield operator value\fR". The field to test is always the left member of the expression.
.LP
The field is a member of the objects returned by the command. For example, a bank account has "balance", "coming" or "label" fields.
.SS The following operators are supported:
.TP
=
Test if object.field is equal to the value.
.TP
!=
Test if object.field is not equal to the value.
.TP
>
Test if object.field is greater than the value. If object.field is date, return true if value is before that object.field.
.TP
<
Test if object.field is less than the value. If object.field is date, return true if value is after that object.field.
.TP
|
This operator is available only for string fields. It works like the Unix standard \fBgrep\fR command, and returns True if the pattern specified in the value is in object.field.
.SS Expression combination
.LP
You can make a expression combinations with the keywords \fB" AND "\fR, \fB" OR "\fR an \fB" LIMIT "\fR.
.LP
The \fBLIMIT\fR keyword can be used to limit the number of items upon which running the expression. \fBLIMIT\fR can only be placed at the end of the expression followed by the number of elements you want.
.SS Examples:
.nf
.B boobank ls \-\-condition 'label=Livret A'
.fi
Display only the "Livret A" account.
.PP
.nf
.B boobank ls \-\-condition 'balance>10000'
.fi
Display accounts with a lot of money.
.PP
.nf
.B boobank history account@backend \-\-condition 'label|rewe'
.fi
Get transactions containing "rewe".
.PP
.nf
.B boobank history account@backend \-\-condition 'date>2013\-12\-01 AND date<2013\-12\-09'
.fi
Get transactions betweens the 2th December and 8th December 2013.
.PP
.nf
.B boobank history account@backend \-\-condition 'date>2013\-12\-01  LIMIT 10'
.fi
Get transactions after the 2th December in the last 10 transactions

.SH COPYRIGHT
Copyright(C) 2016 Julien Veyssier
.LP
For full copyright information see the COPYING file in the weboob package.
.LP
.RE
.SH FILES
"~/.config/weboob/backends" 

"~/.config/weboob/qbooblyrics"

.SH SEE ALSO
Home page: http://weboob.org/applications/qbooblyrics


ChangeLog
Weboob 1.5 (2019-02-27)

	General
	* License change of the core from AGPLv3 to LGPLv3.
	* New humanis module (CapBank, CapBankPockets, CapBankWealth)
	* New nef module (CapBank, CapBankTransfer)
	* New netfinca module ()
	* New sogecartenet module (CapBank)
	* New themisbanque module (CapBank, CapProfile)

	General: Core
	* convert lists to displayable ones
	* rewrite to not require module_name useless parameter
	* use docstrings for fields
	* short explanatory paragraph on the class role
	* small fix
	* fail if "Account.number" is NotLoaded
	* add assertNotEmpty method
	* Fix condition for Python version and pyuic
	* BrowserRedirect's __str__ should display url
	* add parent logger param when instanciating a module
	* getLogger creates children for LoggerAdapter too
	* Include build.mk in distribution

	Applications: boobank
	* Typo.

	Applications: boomoney
	* MEMO now contains the raw information, for a better clarity
	* Made -A option to work and select the specified account only
	* Made slight changes so that debug in VSCode is possible
	* Setting Account type to CHECKING temporarily instead of permanently, to avoid side effect e.g. in societegenerale where history was listing the parent account
	* Handling unicode for python 3 compatibility

	Browser
	* More documentation on login management
	* Documentation for HTMLForm.submit parameters
	* follow RFC 7636 on code_challenge generation
	* pass code_challenge_method param
	* log current auth step

	Browser: Filters
	* add new filter ReplaceEntities

	Capabilities
	* Added Profile and Person attributes
	* Added account owner type 'ASSO' (association)
	* add Recipient.__repr__
	* rename diff_percent and keep a compat alias
	* Added field owner_type on Account

	Documentation
	* need_login calls do_login, not do_logout
	* Add imports in examples from module.rst
	* remove trailing parentheses in ItemElement example

	Modules: afer
	* handle new action needed message

	Modules: amazon
	* manage otp
	* fix locate_browser
	* fix download_document

	Modules: americanexpress
	* Add no account url
	* Fix missing transactions

	Modules: amundi
	* Rename iter_investments => iter_investment

	Modules: anticaptcha
	* handle funcaptcha

	Modules: barclays
	* Ignore history for twin accounts that are not in Euros
	* Corrected iter_accounts to fetch IBANs
	* Add loan and revolving credit
	* Use filter ReplaceEntities in common filter
	* Raise ActionNeeded for questionnaire on the website

	Modules: bibliothequesparis
	* items are present twice when late

	Modules: billetreduc
	* base url changed + support accented months

	Modules: binck
	* Skip new questions page
	* Handle ActionNeeded for password changing

	Modules: bnporc
	* id recipient can change, don't check the full recipient id
	* Retrieve also pending recipients
	* avoid recipient with same iban
	* Handle 'No porftolio to display' message

	Modules: bnppere
	* Handled french and american amounts
	* Added iter_pockets to avoid S2E crash when testing
	* Implemented code to avoid account ID duplicates

	Modules: bolden
	* Diff and Doc attributes may be NotAvailable for investments
	* fix liquidities as label changed
	* fix account valuation diff

	Modules: boursorama
	* handle transfer main page to iter recipient
	* Corrected coming summary filtering
	* Corrected liquidities and investments
	* Corrected issues with immediate card transactions
	* Do not scrape deferred card summaries when not fixed yet
	* Added "Retrait dab" as Transaction.TYPE_WITHDRAWAL

	Modules: bouygues
	* fix #203: sms url got updates

	Modules: bp
	* Handle new history url case
	* Get Savings IBAN
	* handle website unavailable message for pro accounts
	* Fix transfer without iban
	* Fix coming url for "Compte attente"
	* fix differed cards url
	* Fix revolving credit navigation
	* Fix url for iban and coming
	* Correct get RIB parameter

	Modules: caissedepargne
	* NotImplementedError for transfer with SMS
	* Check transfer functionality in pro space
	* Adde "is_error" to CreditCooperatifMarketPage
	* Included Capitalisation contracts in iter_investments
	* Corrected type and balance xpath for LI and Capitalisation
	* Fix regression
	* Fix datepourie
	* fix loans
	* Handle "Récapitulatif de frais bancaires" document
	* update main form
	* Some user have a market account but they have no access to linebourse
	* Fix cnet when card have not coming transaction
	* Handle error when error don't have any title
	* Added BrowserUnavailable when the redirection offrebourse is in error

	Modules: cmb
	* inherit CapBankTransfer like parent cmso

	Modules: cmes
	* Skip user form
	* handle new login url and custom page
	* handle new login url
	* Corrected investments and pockets for CCB supports

	Modules: cmmc
	* inherit CapBankTransfer like parent cmso

	Modules: cmso
	* Allows inexistant json transfer keys
	* check if account can do transfer or not
	* fix internal transfer
	* skip duplicate recipients
	* add new feature: transfer
	* add new feature iter recipient

	Modules: cragr
	* fix iter recipient
	* Avoid crash when redirection to Predica for old website
	* avoid duplicate iban
	* add iter recipient and transfer to api website
	* Set vdate to date for Savings accounts transactions
	* Typed "Escompte commercial" as Loan
	* Added missing balance for revolving credits
	* Typed PRET CEL as Loans
	* Remove assert on account balances
	* Use go_to_account_space() instead of self.contract_page.go()
	* Included "/agriculteur" in the list of URLs
	* Use go_to_account_space() for get_profile and advisor
	* Specified landing URL in the error message if login failed
	* Remove assert False since all error message are handled
	* Limit to 11 digits when posting username
	* Added other wrongpass message
	* Added "entreprise" to all concerned URL()
	* Corrected error message during security check
	* Add regexp to login to prevent server errors
	* Handle ActionNeeded when first online connection

weboob/applications/qbooblyrics/qbooblyrics.py
# -*- coding: utf-8 -*-

# Copyright(C) 2016 Julien Veyssier
#
# This file is part of weboob.
#
# weboob is free software: you can redistribute it and/or modify
# it under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# weboob is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
# GNU Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with weboob. If not, see <http://www.gnu.org/licenses/>.


from weboob.capabilities.lyrics import CapLyrics
from weboob.tools.application.qt5 import QtApplication

from .main_window import MainWindow


class QBooblyrics(QtApplication):
    APPNAME = 'qbooblyrics'
    VERSION = '1.6'
    COPYRIGHT = 'Copyright(C) 2016 Julien Veyssier'
    DESCRIPTION = "Qt application allowing to search song lyrics."
    SHORT_DESCRIPTION = "search lyrics"
    CAPS = CapLyrics
    CONFIG = {'settings': {'backend': '',
                           'maxresultsnumber': '10'
                           }
              }

    def main(self, argv):
        self.load_backends([CapLyrics])
        self.load_config()

        main_window = MainWindow(self.config, self.weboob, self)
        main_window.show()
        return self.weboob.loop()
